src/octopeer-github/main/MainController.ts

```diff
@@ -28,21 +29,55 @@ class MainController {
      * Set up all event handlers in the Chrome API.
      */
     private connectToContentScript() {
+        this.initAllCurrentTabs();
+        this.rehookOnUpdate();
+        this.rehookOnFocusChange();
+        this.listenToDatabaseMessages();
+    }
+
+    /**
+     * On start-up, let all tabs hook to DOM
+     */
+    private initAllCurrentTabs() {
         const self = this;
-        // On start-up, let all tabs hook to DOM
         chrome.tabs.query({}, function(tabs) {
             let tab: Tab;
             for (tab of tabs) {
                 self.testAndSend(tab);
             }
         });
-        // Whenever a tab updates, send a message to re-hook to DOM
+    }
+
+    /**
+     * Whenever a tab updates, send a message to re-hook to DOM
+     */
+    private rehookOnUpdate() {
+        const self = this;
         chrome.tabs.onUpdated.addListener(function(tabId, changeInfo, tab) {
             if (changeInfo.status && changeInfo.status === "complete") {
                 self.testAndSend(tab);
             }
         });
-        // When a tab sends a message, log it to the Database
+    }
+
+    /**
+     * Whenever the user changes tabs, send a message to re-hook to DOM
+     */
+    private rehookOnFocusChange() {
+        const self = this;
+        chrome.tabs.onActivated.addListener(function(activeInfo) {
+            chrome.tabs.query({"active": true, "windowId": activeInfo.windowId}, function (tabs: Tab[]) {
+                // There is usually only one active tab, so no need to iterate
```

Comment:
I think you can move this line to the method documentation, for clarity reasons.